include/mbedtls/sha256.h

```diff
@@ -59,7 +59,17 @@ typedef struct mbedtls_sha256_context {
 mbedtls_sha256_context;
 
 #else  /* MBEDTLS_SHA256_ALT */
+
```

Comment:
One way or the other, it would be good to have a non-regression test. We test alt headers by doing a build (compilation only, no running) with the headers in [`tests/include/alt-dummy`](https://github.com/Mbed-TLS/mbedtls/tree/development/tests/include/alt-dummy). What can we put in these headers to get a compilation error if the C/C++ status is wrong when compiling an application with a C++ compiler?